src/jarallax.esm.js

```diff
@@ -64,9 +65,29 @@ domReady(() => {
 // list with all jarallax instances
 // need to render all in one scroll/resize event
 const jarallaxList = [];
+let globalParallaxUpdateInit = false;
 
 // Animate if changed window size or scrolled page
 let oldPageData = false;
+
+function parents(elem, selector) {
+    const elements = [];
+    const ishaveselector = selector !== undefined;
+
+    while (elem.parentElement !== null) {
+        elem = elem.parentElement;
+        if (elem.nodeType !== Node.ELEMENT_NODE) {
+            continue;
+        }
+
+        if (!ishaveselector || elem.matches(selector)) {
+            elements.push(elem);
```

Comment:
As I understand from your usage of the `parents` function you never care about any other parent than the first hit. We could refactor this to a `parent` function and break the loop after the first match is found.

Reply:
I tried to make it similar to jQuery.parents. Maybe, later, it would be helpful.

Reply:
@omiroshnichenko yup, but the changes are never using something higher than the parent, so it's a waste of processing time to continue looping the rest of the tree.

Reply:
The method has been renamed to `getParentBySelector` and returns the first parent that matches the selector.